src/Microsoft.AspNetCore.Authentication2/DefaultAuthenticationManager.cs

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authentication2
+{
+    public class DefaultAuthenticationManager : IAuthenticationManager2
+    {
+        // TODO: figure out some other way to get the context??
+        public DefaultAuthenticationManager(IAuthenticationSchemeProvider schemes, IHttpContextAccessor context)
+        {
+            Schemes = schemes;
+            _accessor = context;
+        }
+
+        public IAuthenticationSchemeProvider Schemes { get; }
+
+        // handler instance cache, need to initialize once per request, manager lifetime should be per request
+        private Dictionary<string, IAuthenticationSchemeHandler> _handlerMap = new Dictionary<string, IAuthenticationSchemeHandler>();
+
+        private readonly IHttpContextAccessor _accessor;
```

Comment:
Yeah its on the todo list, possibly we modify the signature of all of the methods to explicitly pass in an HttpContext instead.